### PR TITLE
OSDF Collector

### PR DESCRIPTION
**Pull request by bbockelm**
This is a simple container meant to run a HTCondor collector on behalf of the OSDF.

Collector status information in a container is an old idea; we have about 5 of the OSDF cache installs reporting to collector.opensciencegrid.org currently.  This represents a fresh approach where we'll have a combination of topology and the shovelers pushing ads to the collector, then have a dedicated OSDF webapp doing matchmaking between clients and sites.  The long-term goal is to provide an abstraction layer between OSDF and other OSG services so one could setup a standalone OSDF without any other OSG-provided service (turning OSDF more into a software product).

But for now, just a collector.

(Oh, and a collector with rsyslog integration so the container logs "do the right thing")

### CODE REVIEW

**Comment by bbockelm** [comment]
@matyasselmeci - can I get a quick review on this?  It's critical path for Justin's work on a cache manager.

File: opensciencegrid/osdf-collector/etc/supervisord.d/rsyslogd.conf [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[program:rsyslogd]
+directory=/
+command=/usr/sbin/rsyslogd -n
+autorestart=true
+stdout_logfile=/proc/self/fd/1
+stdout_logfile_maxbytes=0
+stderr_logfile=/proc/self/fd/2
+stderr_logfile_maxbytes=0
```
Comment (matyasselmeci) [inline opensciencegrid/osdf-collector/etc/supervisord.d/rsyslogd.conf:8]:
```suggestion
stdout_logfile=/dev/stdout
stdout_logfile_maxbytes=0
stderr_logfile=/dev/stderr
stderr_logfile_maxbytes=0
```

File: opensciencegrid/osdf-collector/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
+ARG BASE_OSG_SERIES=3.6
+ARG BASE_YUM_REPO=testing
+
+FROM opensciencegrid/software-base:${BASE_OSG_SERIES}-el8-${BASE_YUM_REPO}
+LABEL maintainer "OSG Software <help@opensciencegrid.org>"
+
+# Previous args have gone out of scope
+ARG BASE_OSG_SERIES=3.6
+ARG BASE_YUM_REPO=testing
```
Comment (matyasselmeci) [inline opensciencegrid/osdf-collector/Dockerfile:9]:
```suggestion
```
You don't need these, they're not used in the rest of your file.

File: opensciencegrid/osdf-collector/etc/condor/config.d/50-osdf-collector.conf [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+# Forward OSDF ads to the CE collector
+CONDOR_VIEW_HOST=collector.opensciencegrid.org:9619
+UPDATE_VIEW_COLLECTOR_WITH_TCP=True
+
+# Not strictly necessary - but convenient for when you exec into the container:
+COLLECTOR_HOST=localhost
+CONDOR_HOST=localhost
+
+# Improve default logging to match our syslog scripts
+ALL_DEBUG=D_CAT,D_SUB_SECOND,D_PID
```
Comment (matyasselmeci) [inline opensciencegrid/osdf-collector/etc/condor/config.d/50-osdf-collector.conf:41]:
```suggestion
ALL_DEBUG=D_CAT,D_PID
```
The regexps in rsyslog.conf do not appear to match sub-second times.

**Review by matyasselmeci — COMMENTED** [review]
Got a few suggestions but otherwise fine. Let me know when you've made the changes.

**Pull request merged**